CRM/Core/Payment/Form.php

```diff
@@ -65,11 +65,15 @@ static public function setPaymentFieldsByProcessor(&$form, $processor, $forceBil
     $form->assign('paymentTypeName', $paymentTypeName);
     $form->assign('paymentTypeLabel', $paymentTypeLabel);
     $form->_paymentFields = $form->billingFieldSets[$paymentTypeName]['fields'] = self::getPaymentFieldMetadata($processor);
-    $form->_paymentFields = array_merge($form->_paymentFields, self::getBillingAddressMetadata($processor, $form->_bltID));
     $form->assign('paymentFields', self::getPaymentFields($processor));
-    self::setBillingAddressFields($form, $processor);
-    // @todo - this may be obsolete - although potentially it could be used to re-order things in the form.
-    $form->billingFieldSets['billing_name_address-group']['fields'] = array();
+    // return without adding billing fields if billing_mode = 4 (@todo - more the ability to set that to the payment processor)
+    // or payment processor is NULL (pay later)
+    if (CRM_Utils_Array::value('billing_mode', $processor) != 4) {
+      $form->_paymentFields = array_merge($form->_paymentFields, self::getBillingAddressMetadata($processor, $form->_bltID));
+      self::setBillingAddressFields($form, $processor);
```

Comment:
Thanks for reviewing this from your side. I just wanted to clear some points regarding what I'm getting here -

I don't get `billing_fields` to DISPLAY when I select Paypal Standard, but the `billing_fields` are inserted in `$form->_paymentFields` due to this array_merge happening here which leads to [formRule error message](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/Contribution/Main.php#L905) of required field for all the `billing_fields` even if they are not displayed on the page.

Thanks Again.